apps/heft/src/utilities/ConfigFile.ts

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import { ConfigurationFile, InheritanceType, PathResolutionMethod } from '@rushstack/heft-config-file';
+
+import { ICopyStaticAssetsConfigurationJson } from '../plugins/CopyStaticAssetsPlugin';
+import { IApiExtractorPluginConfiguration } from '../plugins/ApiExtractorPlugin/ApiExtractorPlugin';
+import { ITypeScriptConfigurationJson } from '../plugins/TypeScriptPlugin/TypeScriptPlugin';
+import { HeftConfiguration } from '../configuration/HeftConfiguration';
+import { FileSystem } from '@rushstack/node-core-library';
+
+export enum HeftEvent {
+  clean = 'clean',
+  preCompile = 'pre-compile',
+  compile = 'compile',
+  bundle = 'bundle',
+  postBuild = 'post-build'
+}
+
+export interface IHeftConfigurationJsonEventActionBase {
+  actionKind: string;
+  heftEvent: 'clean' | 'pre-compile' | 'compile' | 'bundle' | 'post-build';
+  actionId: string;
+}
+
+export interface IHeftConfigurationDeleteGlobsEventAction extends IHeftConfigurationJsonEventActionBase {
+  actionKind: 'deleteGlobs';
+  globsToDelete: string[];
+}
+
+export interface IHeftConfigurationJsonPluginSpecifier {
+  plugin: string;
+  options?: object;
+}
+
+export interface IHeftConfigurationJson {
+  eventActions?: IHeftConfigurationJsonEventActionBase[];
+  heftPlugins?: IHeftConfigurationJsonPluginSpecifier[];
+}
+
+export interface IHeftEventActions {
+  deleteGlobs: Map<HeftEvent, IHeftConfigurationDeleteGlobsEventAction[]>;
+}
+
+export class ConfigFile {
```

Comment:
It took me some puzzling to figure out that this is not a generic `ConfigFile` loader, nor is it the `heft.json` loader, but rather it's a static class containing the loader functions for various core config files.

It might help to rename it to `CoreConfigFiles`.

Reply:
Doc comments would also be pretty useful.  This is a pretty important source file. Something like this:

```ts
/**
  * Returns the loader for the `config/heft.json` config file.
 */
public static get heftConfigFileLoader(): ConfigurationFile<IHeftConfigurationJson>;

/**
 * Attempts to load the `config/heft.json` config file from its default location.
 */
 public static async tryLoadHeftConfigFileFromDefaultLocationAsync();
```